extension/inet/inet_escape_functions.cpp

```diff
@@ -0,0 +1,491 @@
+		                        invalid_codepoints == other.invalid_codepoints &&
+		                        special_invalid_charrefs == other.special_invalid_charrefs));
+	}
+};
+
+unique_ptr<FunctionData> UnescapeBind(ClientContext &context, ScalarFunction &bound_function,
+                                      vector<unique_ptr<Expression>> &arguments) {
+	return make_uniq<UnescapeBindData>();
+}
+
+static idx_t to_utf8(uint32_t code_point, char arr[4]) {
+	int sz = 0;
+	if (code_point <= 0x7F) {
+		arr[0] = static_cast<char>(code_point);
+		sz = 1;
+	} else if (code_point <= 0x7FF) {
+		arr[0] = static_cast<char>(0xC0 | ((code_point >> 6) & 0x1F));
+		arr[1] = static_cast<char>(0x80 | (code_point & 0x3F));
+		sz = 2;
+	} else if (code_point <= 0xFFFF) {
+		arr[0] = static_cast<char>(0xE0 | ((code_point >> 12) & 0x0F));
+		arr[1] = static_cast<char>(0x80 | ((code_point >> 6) & 0x3F));
+		arr[2] = static_cast<char>(0x80 | (code_point & 0x3F));
+		sz = 3;
+	} else if (code_point <= 0x10FFFF) {
+		arr[0] = static_cast<char>(0xF0 | ((code_point >> 18) & 0x07));
+		arr[1] = static_cast<char>(0x80 | ((code_point >> 12) & 0x3F));
+		arr[2] = static_cast<char>(0x80 | ((code_point >> 6) & 0x3F));
+		arr[3] = static_cast<char>(0x80 | (code_point & 0x3F));
+		sz = 4;
+	} else {
+		throw InternalException("Cannot convert codepoint: out of range ");
+	}
+	return sz;
+};
+
+class Match {
+public:
+	Match(idx_t beg, idx_t len, bool semi) : start(beg), size(len), semicolon(semi), next(nullptr) {
+	}
+
+public:
+	string_t decoded;
+	idx_t start = 0;
+	idx_t size = 0;
+	bool semicolon = false;
+	Match *next = nullptr;
+};
+
+struct UnicodeValue {
+	static bool check_and_cast_input(Match *match, const char *input_data, int32_t &num,
+	                                 idx_t &num_of_chars_to_decode) {
+		auto start = match->start;
+		auto size = match->size;
+		for (idx_t i = start + 1; i < start + size; i++) {
+			int digit = Blob::HEX_MAP[static_cast<unsigned char>(input_data[i])];
+			if (digit < 0 || 10 < digit) { // unicode points should not have character digits
+				num_of_chars_to_decode = i - start - 1;
+				return false;
+			}
+			num *= 10;
+			num += digit;
+		}
+		return false;
+	}
+};
+
+struct HexValue {
+	static bool check_and_cast_input(Match *match, const char *input_data, int32_t &num,
+	                                 idx_t &num_of_chars_to_decode) {
+		if (is_control_point(match, input_data)) {
+			return true;
+		}
+		auto start = match->start;
+		auto size = match->size;
+		for (idx_t i = start + 2; i < start + size; i++) {
+			int hex_digit = Blob::HEX_MAP[static_cast<unsigned char>(input_data[i])];
+			if (hex_digit < 0) { // true if it's not a valid hexadecimal digit
+				num_of_chars_to_decode = i - start - 1;
+				return false;
+			} else {
+				// overflow - hexadecimal value too large to fit in an 32-bit int
+				if (num * 16 > (INT32_MAX - hex_digit) / 16) {
```

Comment:
Perhaps we can use `TryDecimalMultiply::Operation` since it's specialized for this

I'm a little scared that `num * 16` needs a cast here to do this check properly, relying on TryDecimalMultiply seems easier than doing it by hand